### ISSUE TITLE
Intel AMT Setup

### ISSUE THREAD

**Issue by PathfinderNetworks** [Ylianst/MeshCentral#2359]
I know there have been multiple issues opened asking similar questions- so I apologize if this is a repeat.  
I don't know much about Intel AMT but am trying to wrap my head around it.  
I have a brand new HP EliteDesk here at my office that I'm preparing for a client.  It supports AMT.  
I created the setup.bin file and dropped it on a USB flash drive and booted the PC with it.  It did pull down the configuration from the flash drive and configured AMT on the system. I can tell it did this due to the password I created during the setup.bin configuration now being present for logging in to the Intel ME setup on the device.
I can also see, in MeshCentral, that the device is showing that Intel ME is activated in ACM mode. 
However, I can't actually connect to the device using any of the AMT functions. All the options are greyed out. 

When I open the Intel ME console on the device itself and look at the Provisioning Record I noticed something- something that may be causing these issues?
For the FQDN record it is showing 'rootcert.meshcentral.com'.  
Shouldn't the FQDN be pointing to my own MeshCentral server? 

I've attached an image of the configuration screen of this device. 
![20210307_151319](https://user-images.githubusercontent.com/48607498/110253413-2f290880-7f58-11eb-9c84-c753803fc322.jpg)

**Comment by PathfinderNetworks** [comment]
Also, if I open the setup.bin file with a text editor I can see the same 'rootcert.meshcentral.com' data being present.

**Comment by Ylianst** [comment]
Nice. You did everything right. The "rootcert.meshcentral.com" name is a dummy name used for USB setup, I should really change it to use the DNS name of your server, it would make a lot more sense. I will fix that.

In any case, your computer is in ACM which is great. If you setup a Intel AMT policy that uses a randomized password, you will find the password that MeshCentral setup in the "amtactivation.txt" file in "meshcentral-data".

For grayed out features. If you MeshCentral server is in "WAN mode", the only way for these features to work is to have Intel AMT connect back to the server using CIRA (Client Initiated Remote Access). Generally, your server should have an exposed port 4433 for Intel AMT to connect to. You can try https://youserver:4433 to test if the port is open. It's going to be an un-trusted TLS cert so the browser will give you a warning, but if you ignore it you will get a text message indicating this server is for Intel AMT. If you are running behind a reverse proxy, make sure to forward TCP connections directly to this port. Don't try to TLS offload or change the certificate, AMT expects the exact TLS certificate that MeshCentral is offering. Once AMT connects, you will see "CIRA" on the device and options will light up.

If you have in LAN mode or Hybrid mode and your computer is on the local network, it's possible for MeshCentral to connect to your Intel AMT device. Of course, this depends on the policy you set.

Once you got AMT running, you can do a bunch of operations on a device even if the agent is not running or the computer is soft-off.

Hope that helps,
Ylian

**Comment by PathfinderNetworks** [comment]
Thanks Ylian,
I do have that port open and get the "MeshCentral MPS server.
Intel® AMT computers should connect here." message.

My server is in WAN mode.  The computer I set this up on is currently connected to the same LAN the MeshCentral server is on- so that may explain why it's not making the CIRA connection.
I'm going to be installing the PC at the client site tomorrow. So, hopefully, that will allow it to connect back to my MeshCentral server with CIRA and the AMT features will be available.
I've got other devices that are just activated in CCM mode and they do connect (although you can't really do much other than view data and power cycle the system in CCM).  I wish it were easier to setup Intel AMT in ACM mode. The USB flash drive option isn't difficult- it's just that I don't always have direct physical access to the device to set it up via USB.  I wish it were possible to get ACM mode working  via the MeshCentral agent.

**Comment by Ylianst** [comment]
FYI. Just published MeshCentral v0.7.83 that creates a setup.bin file with your server's DNS name. Old setup.bin file will still work, but this is a lot better.

For CIRA or any Intel AMT connection, make sure that a "managed network interface" is used. That is, if a device has two Ethernet ports, you have to make sure the port that is managed by Intel AMT is connected to the Internet. This is because Intel AMT will bypass the OS and have it's own drivers and is physically connected to one Ethernet port and sometimes a WIFI interface. Intel AMT has only one WIFI driver, so you need a specific WIFI adapter plugged in a specific way in the system. SO, if you connect to the Internet using a USB adapter or something else, Intel AMT will not be able to connect.

As for activation into ACM, the difficulty is due to security. You need to prove physical access so not to allows any malware to activate ACM. There is an option to buy a specific activation certificate with a DNS name like "example.com" and if you load that cert into MeshCentral and the Intel AMT device Ethernet port is connected to a DHCP server that indicates "example.com" as the local domain, MeshCentral can activate into ACM. This ACM activation system does not work over WIFI.

**Comment by PathfinderNetworks** [comment]
Thanks Ylian,
There is still something not quite right with these AMT connections.  I installed this new PC at the customer site today. It only has a single ethernet port so Intel AMT is running over the same ethernet connection as is Windows.  I can see that Intel AMT pulls an IP address via DHCP.  
However, it still will not create a CIRA connection to my MeshCentral server.  I've noticed that same issue with other Intel AMT devices.  Some do establish a CIRA connection and others do not.  There seems to be no rhyme or reason why some do and some don't.  
The ones that have a CIRA connection I can perform AMT functions.  Those that don't show a CIRA connection the AMT functions are all greyed out.

**Comment by jsastriawan** [comment]
There are several potential issues if AMT CIRA does not connect. Some AMT devices especially that were built by some OEM does not have a unique UUID. The UUID is the identifier meshcentral use to track CIRA connection. You could try using amt command on agent console to see if the UUID is not unique. If it is not unique, you could regenerate it using BIOS vendor tool like AMIDE[DOS|WIN64] etc.

**Comment by PathfinderNetworks** [comment]
So this is strange.  I updated my MeshCentral server to 0.7.88.  After I did that this particular PC I was discussing made the CIRA connection and is now fully working with Intel AMT.  Nothing on the computer itself changed.  So I'm not sure if the act of updating the server resolved it or if just restarting the server did it. Whatever the case the only component that changed was updating/restarting the MeshCentral server.
